### PR TITLE
Fix multiple user login problem

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
There was an issue where only one user could login to the EI back end using the front-end. This was caused by the front-end reusing the same clienbt for each call, and when a user did login it was the clients session that became logged in, and all users of a front-end used the same client.

### Applicable Issues
Only one user could be logged in at the same time.
Mockito was not working well in windows.

### Description of the Change
Login is no longer bound the the bridge client, but a new client is createt for each bridged calls. An X-Auth-Token is saved for each users session in the bridge and is reused to authenticate against the back end for as long as the user is logged in.

Tests uses mockserverclient instead if Mockito for handeling http request calls in tests.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/resources/static/js/subscription.js [hunk 1]
```diff
@@ -552,11 +551,10 @@ jQuery(document).ready(function () {
                 reload_table();
             },
             error: function (XMLHttpRequest, textStatus, errorThrown) {
-                window.logMessages(XMLHttpRequest.responseText);
                 reload_table();
                 var responseJSON = JSON.parse(XMLHttpRequest.responseText);
                 for (var i = 0; i < responseJSON.length; i++) {
-                    $.jGrowl(responseJSON[i].subscription + " :: " + responseJSON[i].reason, { sticky: true, theme: 'Error' });
+                    window.logMessages("Error deleteing subscription: [" + responseJSON[i].subscription + "] Reson: [" + responseJSON[i].reason + "]");
```
Comment (Christoffer-Cortes) [inline src/main/resources/static/js/subscription.js]: 'Reason'

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java [hunk 2]
```diff
@@ -48,8 +52,8 @@
 public class EIRequestsController {
 
     private static final Logger LOG = LoggerFactory.getLogger(EIRequestsController.class);
-
-    private CloseableHttpClient client = HttpClientBuilder.create().build();
+    private static final String X_PATH_NAME = "x-auth-token";
```
Comment (Christoffer-Cortes) [inline src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Couldn't this simply be called X_AUTH or X_AUTH_TOKEN?

Comment (tobiasake) [reply src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Is "x-auth-token" HTTP Standard?
I think also that we should have "auth" in the varaible name.

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java [hunk 3]
```diff
@@ -64,13 +68,10 @@
...
         HttpGet eiRequest = new HttpGet(eiRequestUrl);
-        String header = request.getHeader("Authorization");
 
-        if (header != null) {
-            eiRequest.addHeader("Authorization", header);
-        }
+        eiRequest = (HttpGet) addHeadersToRequest(eiRequest, request);
 
-        return getResponse(eiRequest);
+        return getResponse(eiRequest, request);
```
Comment (tobiasake) [inline src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Why two different two requests?
Don't add headers to the existing request?

Comment (tobiasake) [reply src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Same comment on the other places in the code where this function is called.

File: src/main/resources/static/js/subscription.js [hunk 4]
```diff
@@ -1090,7 +1088,8 @@ jQuery(document).ready(function () {
                 var errors = "";
                 for (var i = 0; i < responseJSON.length; i++) {
                     errors = errors + "\n" + responseJSON[i].reason;
-                    $.jGrowl(responseJSON[i].subscription + " :: " + responseJSON[i].reason, { sticky: true, theme: 'Error' });
+                    // Errors are displayed in the form.
+                    // $.jGrowl(responseJSON[i].subscription + " :: " + responseJSON[i].reason, { sticky: true, theme: 'Error' });
```
Comment (tobiasake) [inline src/main/resources/static/js/subscription.js]: Remove this commented out code if it is not going to be used.

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java (hunk 2 above)
Comment (Christoffer-Cortes) [reply src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: No, it's an unregistered header and is missing here: https://www.iana.org/assignments/message-headers/message-headers.xhtml.
It's used in spring for session authentication and every other app can have their own definition of its use.

File: src/main/resources/static/js/subscription.js (hunk 1 above)
Comment (a-breid-ericsson) [reply src/main/resources/static/js/subscription.js]: Fixed.

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java (hunk 2 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Fixed.

File: src/main/java/com/ericsson/ei/frontend/EIRequestsController.java (hunk 3 above)
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/frontend/EIRequestsController.java]: Fixed.

File: src/main/resources/static/js/subscription.js (hunk 4 above)
Comment (a-breid-ericsson) [reply src/main/resources/static/js/subscription.js]: Fixed.

**Pull request merged**